src/librustc_typeck/diagnostics.rs

```diff
@@ -3426,6 +3426,50 @@ parameters. You can read more about it in the API documentation:
 https://doc.rust-lang.org/std/marker/struct.PhantomData.html
 "##,
 
+E0436: r##"
+A struct can include `..` to indicate that you want to use a copy of some other
+struct for some of the values. This functional update syntax can only be used
+on struct types. It is an error to use it on other types, such as enums.
+
+Example of erroneous code:
+
+```compile_fail
+enum Foo {
+    A { x: u32, y: u32 }
+}
+
+fn bar() {
+    baz(Foo::A { x: 5, y: 10 });
+}
+
+fn baz(Foo f) -> Foo {
+    Foo::A { x: 6, ..f } // error: functional record update syntax requires
+                         //        a struct
+}
+```
+
+To fix this, you need to manually initialize the value. For enums, you should
+match on the variants and copy the values to the new enum value:
+
+```
+enum Foo { A { x: u32, y: u32 } }
```

Comment:
However, this one is still incorrect (the enum declaration should be on one line).